Reference/google/protobuf/any.pb.swift

```diff
@@ -108,18 +108,14 @@ import Foundation
 ///         "@type": "type.googleapis.com/google.protobuf.Duration",
 ///         "value": "1.212s"
 ///       }
-struct Google_Protobuf_Any: ProtobufGeneratedMessage, ProtobufProto3Message {
+struct Google_Protobuf_Any: ProtobufGeneratedMessage, ProtobufProto3Message, ProtoNameProviding {
   public var swiftClassName: String {return "Google_Protobuf_Any"}
   public var protoMessageName: String {return "Any"}
   public var protoPackageName: String {return "google.protobuf"}
-  public var jsonFieldNames: [String: Int] {return [
-    "typeUrl": 1,
-    "value": 2,
-  ]}
-  public var protoFieldNames: [String: Int] {return [
-    "type_url": 1,
-    "value": 2,
-  ]}
+  public static let _protobuf_fieldNames: FieldNameMap = [
+    1: .unique(proto: "type_url", json: "typeUrl", swift: "typeURL"),
+    2: .same(proto: "value", swift: "value"),
```

Comment:
Features are nice if they provide high value, but it's hard for me to see "the field is rendered as `fooBar` instead of `foo_bar`" as a high value feature, at least not worth the complexity it would add to the name bundles (what if proto name, JSON name, and Swift name are all different? What if proto name and Swift name are the same but JSON name is different? What if...). And aside from the debugger and playgrounds, are there any other compelling use cases for `Mirror` that would be harmed by using the proto names instead of the Swift ones? (To be honest, I've never been a fan of that API. What looks at first glance like it's supposed to be a general reflection API turns out to be nothing more than a value snapshot renderer that hooks into limited places in the IDE.)

One of the major design goals for protocol buffers in general has been to make sure we're not shipping anything that users don't truly need—that's why, for example, the Objective-C implementation doesn't generate descriptors. We still don't have the luxury of effectively-infinite space on user's phones like we do on servers, and I want to stay very conscious of that since mobile is a major audience for this library. Sure, if we look at individual strings here and there it's easy to say "that's only a few bytes, what's the big deal?" but if we add up a lot of those, we start to get real savings.

So if it seems like I'm harping on about a handful of characters, it's more of a general design philosophy than picking at specific nits. IMO, we really must be focusing on performance and size right now instead of minor features in this early period before we're doing official releases and we have the "we will break you" warning banner up. It's **much** harder to take away or change things once people are depending on them than it is to add new things later.